modules/tinymce/src/plugins/table/test/ts/browser/command/ModifyClassesCommandsTest.ts

```diff
@@ -0,0 +1,249 @@
+import { context, describe, it } from '@ephox/bedrock-client';
+import { TinyHooks, TinySelections } from '@ephox/mcagar';
+import { PlatformDetection } from '@ephox/sand';
+import { assert } from 'chai';
+import Editor from 'tinymce/core/api/Editor';
+import { EditorEvent } from 'tinymce/core/api/util/EventDispatcher';
+import { TableModifiedEvent } from 'tinymce/plugins/table/api/Events';
+import Plugin from 'tinymce/plugins/table/Plugin';
+import Theme from 'tinymce/themes/silver/Theme';
+
+const platform = PlatformDetection.detect();
+
+describe('browser.tinymce.plugins.table.command.ModifiyClassesCommandsTest', () => {
+  const hook = TinyHooks.bddSetupLight<Editor>({
+    plugins: 'table',
+    indent: false,
+    valid_styles: {
+      '*': 'height,vertical-align,text-align,float,border-color,background-color,border,padding,border-spacing,border-collapse'
+    },
+    base_url: '/project/tinymce/js/tinymce',
+    setup: (editor: Editor) => {
+      editor.on('tablemodified', logEvent);
+    }
+  }, [ Plugin, Theme ], true);
+
+  const cleanTableHtml = (html: string) =>
+    html.replace(/<p>(&nbsp;|<br[^>]+>)<\/p>$/, '');
+
+  let events: Array<EditorEvent<TableModifiedEvent>> = [];
+  const logEvent = (event: EditorEvent<TableModifiedEvent>) => {
+    events.push(event);
+  };
+
+  const execCmdAndAssertEvent = (label: string, editor: Editor, cmdName: string, ui?: boolean, data?: string) => {
+    assert.lengthOf(events, 0, label + ', events not zero');
+    editor.execCommand(cmdName, ui, data);
+    assert.lengthOf(events, 1, label + ', events not one');
+    assert.equal(events[0].type, 'tablemodified', label + ', event not tablemodified');
+    assert.isTrue(events[0].structure, label + ', structure is not true');
+    assert.isFalse(events[0].style, label + ', style is not false');
+    events = [];
+  };
+
+  context('TINY-7163: Modify cell class', () => {
+    context('Can use mceTableCellToggleClass to toggle the cell class', () => {
+      context('When there is no class already on the cell', () => {
+        const getContentWithoutClass = () => {
+          return (
+            '<table>' +
+              '<tbody>' +
+                '<tr>' +
+                  (platform.browser.isIE() ? '<td>1</td>' : '<td class="">1</td>') +
```

Comment:
Why is this needed? This smells of a bug being somewhere, as we should have `class=""` if we're testing without a class. Additionally, if we do want it then it should be the same on all browsers for something like this.

Reply:
If you remember I do believe I asked you specifically about this. Short version, `class.toggle` leaves behind the `class=""` even though it shouldn't (As `class.remove` does not leave behind this empty string), but IE used a different method which doesn't. It's unrelated to this issue.

Reply:
Ahh, yep I do recall that now sorry. We should fix that at some stage, but as we said for now it's fine/out of scope 👍